### PR TITLE
fix(codewhisperer): closing paren

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Problem

https://github.com/aws/aws-toolkit-vscode/assets/96078566/184b4b2c-0101-48aa-b8d7-81e87d2d6c2c

https://github.com/aws/aws-toolkit-vscode/assets/96078566/0f616265-54bc-453b-a66e-5dc4c844211b

https://github.com/aws/aws-toolkit-vscode/assets/96078566/2fefd6fc-805d-4594-b7b1-f3b7eff07347

https://github.com/aws/aws-toolkit-vscode/assets/96078566/eeef3c13-8a92-4ef0-a855-a69d0900b2eb

https://github.com/aws/aws-toolkit-vscode/assets/96078566/605ff5f0-27f8-4d25-bc1a-eb08d41cf8ad

https://github.com/aws/aws-toolkit-vscode/assets/96078566/cd8f1f4c-90a4-4a86-9ecc-bfefe35a14da

https://github.com/aws/aws-toolkit-vscode/assets/96078566/73711159-fb6e-4e1c-81de-f36739b596c7

https://github.com/aws/aws-toolkit-vscode/assets/96078566/51028f34-ecd2-49e9-91b1-ee7d35b99d56

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/lint/testLint.ts
Comment (justinmk3) [inline scripts/lint/testLint.ts:8]: ?

**Comment by justinmk3** [comment]
Is there a customer facing change that should be mentioned in the changelog?

    npm run newChange

File: scripts/lint/testLint.ts
Comment (Will-ShaoHua) [reply scripts/lint/testLint.ts:8]: will revert this, not sure where is this change coming from

**Comment by Will-ShaoHua** [comment]
@justinmk3 yes I will add change log item for this change since it's fixing an issue broadly reported by customers. We will finalize this change once our bugbash today looks all good

File: scripts/lint/testLint.ts
Comment (Will-ShaoHua) [reply scripts/lint/testLint.ts:8]: @justinmk3 this change seems to be made by linter automatically when I make a commit

File: .changes/next-release/Bug Fix-6847712a-d3eb-46d2-b57e-711715f3cff9.json
Comment (justinmk3) [inline .changes/next-release/Bug Fix-6847712a-d3eb-46d2-b57e-711715f3cff9.json:3]:
```suggestion
	"description": "CodeWhisperer: fix and improve closing symbols handling (brackets, parenthesis, quotes etc)"
```

File: src/codewhisperer/util/closingBracketUtil.ts
Comment (justinmk3) [inline src/codewhisperer/util/closingBracketUtil.ts:236]: This is getting to be an extreme amount of code for non-trivial but very common text-editing operations. I.e. it's time to consider looking at a NPM library that can do these calculations instead of re-inventing them.

Comment (Will-ShaoHua) [reply src/codewhisperer/util/closingBracketUtil.ts:236]: @justinmk3 I will take this as a follow up, let me create a tt as CW tech debt and we will be tracking there

Comment (Will-ShaoHua) [reply src/codewhisperer/util/closingBracketUtil.ts:236]: since this issue is broadly raised by customers, we want to roll out this ASAP lol

**Review by zixlin7 — APPROVED** [review]

File: src/codewhisperer/util/closingBracketUtil.ts
Comment (andrewyuq) [inline src/codewhisperer/util/closingBracketUtil.ts:25]: Looks like this can be derived from the openning map so you don't need to maintain both.

File: src/codewhisperer/util/closingBracketUtil.ts
Comment (andrewyuq) [inline src/codewhisperer/util/closingBracketUtil.ts:27]: `opening`

File: src/codewhisperer/util/closingBracketUtil.ts
Comment (andrewyuq) [inline src/codewhisperer/util/closingBracketUtil.ts:52]: Can the keys be derived from the `opening` map? Otherwise you need to make changes to both places.

**Review by andrewyuq — APPROVED** [review]

File: src/codewhisperer/util/closingBracketUtil.ts
Comment (andrewyuq) [inline src/codewhisperer/util/closingBracketUtil.ts:29]: `parenthesis` does not have `<` and `openToCloseParen` has `<`, is this intended?

Comment (Will-ShaoHua) [reply src/codewhisperer/util/closingBracketUtil.ts:29]: oh no thanks for catching this

Comment (Will-ShaoHua) [reply src/codewhisperer/util/closingBracketUtil.ts:29]: will attach some tests gif in the PR

Comment (Will-ShaoHua) [reply src/codewhisperer/util/closingBracketUtil.ts:29]: this should be finalized now,  i think it already solve most of scenarios

**Review by justinmk3 — APPROVED** [review]
waiting for cw team approval

**Comment by Will-ShaoHua** [comment]
> waiting for cw team approval

yea we're green to release this PR this week, thanks Justin

**Pull request merged**
